java-checks/src/main/java/org/sonar/java/checks/IndentationAfterConditionalCheck.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.BlockTree;
+import org.sonar.plugins.java.api.tree.ForEachStatement;
+import org.sonar.plugins.java.api.tree.ForStatementTree;
+import org.sonar.plugins.java.api.tree.IfStatementTree;
+import org.sonar.plugins.java.api.tree.StatementTree;
+import org.sonar.plugins.java.api.tree.SyntaxTrivia;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.WhileStatementTree;
+
+@Rule(key = "S3973")
+public class IndentationAfterConditionalCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  private JavaFileScannerContext context;
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    this.context = context;
+    scan(context.getTree());
+  }
+
+  @Override
+  public void visitIfStatement(IfStatementTree tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    if (checkElseStatementReports(tree)) {
+      context.reportIssue(this, tree.elseKeyword(),
+        "Use curly braces or indentation to denote the code conditionally executed by this \"" + tree.elseKeyword().text() + "\".");
+    }
+    report(tree.ifKeyword(), tree.closeParenToken(), tree.ifKeyword().text(), findSecondariesToReport(tree));
+    super.visitIfStatement(tree);
+  }
+
+  @Override
+  public void visitWhileStatement(WhileStatementTree tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.whileKeyword(), tree.closeParenToken(), tree.whileKeyword().text(), findSecondariesToReport(tree));
+    super.visitWhileStatement(tree);
+  }
+
+  @Override
+  public void visitForStatement(ForStatementTree tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.forKeyword(), tree.closeParenToken(), tree.forKeyword().text(), findSecondariesToReport(tree));
+    super.visitForStatement(tree);
+  }
+
+  @Override
+  public void visitForEachStatement(ForEachStatement tree) {
+    if (thereIsNoIssue(tree)) {
+      return;
+    }
+    report(tree.forKeyword(), tree.closeParenToken(), tree.forKeyword().text(), findSecondariesToReport(tree));
+    super.visitForEachStatement(tree);
+  }
+
+  private static boolean thereIsNoIssue(Tree tree) {
```

Comment:
Ok. I tend to prefer concise naming so that code stays clear to read, so I would have simply named the method `noIssue`, but it is personal preferences